src/transformStateWithClones.js

```diff
@@ -7,7 +7,42 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let res = [];
+  const newState = { ...state };
+
+  res[0] = newState;
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
```

Comment:
Better use for... of loop to avoid index access
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of
```suggestion
  for (const action of actions) {
    switch (action.type) {
```